### DISCUSSION TITLE
Anyone ever get "unrecognized instruction mnemonic, did you mean: casl?" when building iOS app

### DISCUSSION THREAD

**Discussion by needledrag** [dotnet/maui#15069] [Q&A] [answered]
This is the first time trying to run/build the iOS version of the project as I've only just setup the provisioning profiles.

/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.2.2054/targets/Xamarin.Shared.Sdk.targets(3,3): Error: clang exited with code 1:
obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries/aot-output/arm64/System.Private.CoreLib.dll.s:132:27: error: unexpected token in argument list
mov mono_aot_corlib_got+80(%rip), %rax
                          ^
obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries/aot-output/arm64/System.Private.CoreLib.dll.s:136:1: error: unrecognized instruction mnemonic, did you mean: casl?
call _p_12
^
obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries/aot-output/arm64/System.Private.CoreLib.dll.s:149:27: error: unexpected token in argument list
mov mono_aot_corlib_got+80(%rip), %rax
                          ^
obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries/aot-output/arm64/System.Private.CoreLib.dll.s:153:1: error: unrecognized instruction mnemonic, did you mean: casl?

I can't seem to find much on it so was wondering if it's perhaps a config setting somewhere? 
I'm using XCode 14.2 and am able to build for iOS on a blank project.

Any help or pointing in the right direction would be appreciated.

**Comment by jp6y2009** [comment]
Same issue here. I getting the same error message but that works fine if I changed device to my iPhone (not simulator).
I got this error after updated Visual Studio for Mac.
![image](https://github.com/dotnet/maui/assets/70317128/25c44839-0c44-437b-8ea7-ceb9dce36ece)

**Reply by needledrag** [reply]
The only way I got it to compile is by changing ARM64
to
	  <MtouchArch>x86_64</MtouchArch>

Stopped getting the error messages, not sure it's the solution though

**Reply by jp6y2009** [reply]
![image](https://github.com/dotnet/maui/assets/70317128/1583218d-72f4-40e5-934b-e4ea2bd76fe6)
Did you mean MtouchArch?

**Reply by needledrag** [reply]
Yep

**Reply by jp6y2009** [reply, marked as answer]
According to Rolf's solution, I removed MtouchArch property is worked.
